theia-python-docker/Dockerfile

```diff
@@ -1,22 +1,123 @@
-ARG NODE_VERSION=10
-FROM node:${NODE_VERSION}-stretch
+FROM ubuntu:18.04
 
+ENV DEBIAN_FRONTEND noninteractive
+
+# Common deps
+RUN apt-get update && apt-get -y install curl xz-utils wget gpg
+
+# Install node and yarn
+# From: https://github.com/nodejs/docker-node/blob/6b8d86d6ad59e0d1e7a94cec2e909cad137a028f/8/Dockerfile
+
+# gpg keys listed at https://github.com/nodejs/node#release-keys
+RUN set -ex \
+    && for key in \
+    4ED778F539E3634C779C87C6D7062848A1AB005C \
+    B9E2F5981AA6E0CD28160D9FF13993A75599653C \
+    94AE36675C464D64BAFA68DD7434390BDBE9B9C5 \
+    B9AE9905FFD7803F25714661B63B535A4C206CA9 \
+    77984A986EBC2AA786BC0F66B01FBB92821C587A \
+    71DCFD284A79C3B38668286BC97EC7A07EDE3FC1 \
+    FD3A5288F042B6850C66B31F09FE44734EB7990E \
+    8FCCA13FEF1D0C2E91008E09770F7A9A5AE15600 \
+    C4F0DFFF4E8C1A8236409D08E73BC641CC11F4C8 \
+    DD8F2338BAE7501E3DD5AC78C273792F7D83545D \
+    A48C2BEE680E841632CD4E44F07496B3EB3C1762 \
+    ; do \
+    gpg --batch --keyserver ipv4.pool.sks-keyservers.net --recv-keys "$key" || \
```

Comment:
Do we absolutely need to add the gpg keys, is teher another solution not depending on those keys?

Reply:
This was copied over from the `theia-full-docker` image and I did not want to mess with the setup.

Reply:
Why not stating from another Dockersetup like java0docker !!!

Reply:
It looks like the [node:$NODE_VERSION](https://hub.docker.com/_/node/) docker images, weve been using in other Theia images of this repo, are Debian-based. For example image `node:10` is built from [this Dockerfile](https://github.com/nodejs/docker-node/blob/46f5203674c748b0701135c8eeea4b250b3ecb6d/10/stretch/Dockerfile)

Debian is close to Ubuntu (or the other way around), sharing the `.deb` package format. However the packages are not inter-operable. I think Ubuntu has the greater library of packages, specially with PPAs. 

So for our bigger images like `theia-full` I'm not sure we want to switch to the `node:$version` image as base (we could try as experiment). But this image here looks like it could work without too much trouble. @vince-fugnitto want to give it a try?

Reply:
I can definitely give it a try :)